source/common/filesystem/win32/filesystem_impl.cc

```diff
@@ -19,6 +19,36 @@
 namespace Envoy {
 namespace Filesystem {
 
+namespace {
+static const absl::flat_hash_map<std::string, DWORD> unix_output_pipes_to_windows_standard_handle{
+    {"/dev/stdin", STD_INPUT_HANDLE},       {"/dev/fd/0", STD_INPUT_HANDLE},
+    {"/proc/self/fd/0", STD_INPUT_HANDLE},
+
+    {"/dev/stdout", STD_OUTPUT_HANDLE},     {"/dev/fd/1", STD_OUTPUT_HANDLE},
+    {"/proc/self/fd/1", STD_OUTPUT_HANDLE},
+
+    {"/dev/stderr", STD_ERROR_HANDLE},      {"/dev/fd/2", STD_ERROR_HANDLE},
+    {"/proc/self/fd/2", STD_ERROR_HANDLE}};
```

Comment:
rather than this map, why don't we just add new APIs to the FileSystem class that return already-opened stdio handles. No need to make extra OS calls for this.

Also, why do we need to open new handles to stdio at all in the proxy?

Reply:
>  why don't we just add new APIs to the FileSystem class that return already-opened stdio handles. No need to make extra OS calls for this.

Can you explain a bit more what you mean here?

Do you mean a function added in the `FileSystem` API  like  `handle_t getStandardOutputHandle()`?

> Also, why do we need to open new handles to stdio at all in the proxy?

We dont open new handles, but for the standard handles the OS is the owner of those handle. So we are asking the OS for the latest handle and the OS increments the ref count.

Reply:
> Do you mean a function added in the FileSystem API like handle_t getStandardOutputHandle()?

Yeah, to be more specific, see `include/envoy/filesystem/filesystem.h`. My suggestion would be to add:
```
FIle& standardInput() PURE;
FIle& standardOutput() PURE;
```
to class `Instance`. It would return those descriptors in an open state, and the reference would be owned by `Instance`. That's my proposal, anyway. I think the implementation of each of these would be trivial in both windows and posix.

I'm not sure if this captures what usage model you have in mind. What usage model do you have in mind for these, anyway?

Reply:
cross linking in case anyone else needs it: https://github.com/envoyproxy/envoy/issues/13964

tldr: support file access logging to stdout/err to emulate Linux/macos etc.